src/utils/hibp.js

```diff
@@ -147,8 +155,32 @@ async function loadBreachesIntoApp (app) {
   log.info('done-loading-breaches', 'great success 👍')
 }
 
+async function downloadBreachIcons (breaches) {
+  const breachDomains = breaches.map(breach => breach.Domain)
+  const logoFolder = resolve(dirname(fileURLToPath(import.meta.url)), '../client/images/logo_cache/')
+  try {
+    await mkdir(logoFolder)
+  } catch {
+    // Do nothing; if the directory already exists, that's fine.
+  }
+  breachDomains.forEach(breachDomain => {
+    get(`https://icons.duckduckgo.com/ip3/${breachDomain}.ico`, (response) => {
+      if (response.statusCode !== 200) {
+        return
+      }
+
+      const file = createWriteStream(resolve(logoFolder, breachDomain.toLowerCase() + '.ico'))
+      response.pipe(file)
+      file.on('finish', () => {
+        file.close()
+      })
```

Comment:
Is it possible to add this path to the `breaches` object under the associated breach, to avoid having to call it dynamically for each page render in the controller?

Within `breaches` there exists `breach.LogoPath` which is I believe is unused (and potentially includes an incorrect path) – maybe we can repurpose it?

cc @mansaj for having worked on this last

Reply:
Yeah, I was contemplating doing that, but with the `breaches` object coming directly from HIBP as I understand it, I didn't want to mess with that object. I'm not entirely opposed to that method though.

Another approach would be to still generate a breach-logo mapping, but do it at breach-loading time rather than in the controller, avoiding the delay there. I'll add that tomorrow.

Reply:
Well, it's past midnight here, so officially tomorrow: 80d8f1e7